tests/management/role/test_view.py

```diff
@@ -215,6 +223,19 @@ def test_create_role_allow_list_fail(self):
         response = self.create_role(role_name, in_access_data=access_data)
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
 
+    def test_create_role_appfilter_structure_fail(self):
+        """Test that we cannot create a role with invalid structure of resource definition."""
```

Comment:
Excellent. This is something that is difficult to do with the generated clients.